backend/ingestion/extract_demographic_data.py

```diff
@@ -0,0 +1,206 @@
+"""
+OBJECTIVE: Extract socio-demographic data at the census tract and block level.
+AUTHOR: Benjamin Leiva
+DATE: 04/17/2024
+
+QUERY LIMITS: 50 variables per query, 500 queries per day
+
+CITIES (County, FIPS):
+    - New York:
+        - Manhattan (New York, 36061)
+        - Brooklyn (Kings, 36047)
+        - Queens (Queens, 36081)
+        - Bronx (The Bronx, 36005)
+        - Staten Island (Richmond, 36085)
+    - Chicago:
+        - Chicago (Cook, 17031)
+    - Portland:
+        - Portland (Multnomah, 41051)
+
+SOURCES (Geography Level, Hierarchy):
+    - Census 2020 (state > county > tract)
+        - https://api.census.gov/data/2020/dec/ddhca.html
+        - Variables: Total population, sex, age (by groups is available)
+    - ACS 2022 5Y: (150, state > county > tract > block group)
+        - https://api.census.gov/data/2022/acs/acs5/geography.html   
+        - https://api.census.gov/data/2022/acs/acs5/variables.html
+    - PDB 2022 : (_, state > county > tract > block group)
+        - http://api.census.gov/data/2022/pdb/tract
+        - https://api.census.gov/data/2022/pdb/blockgroup/variables.html
+    - Census Documentation:
+        - https://census-docs.com/#get_function
+
+VARIABLES (ID, source):
+    - Total Population:
+        - B01003_001E, ACS 2022 5Y
+    - Median Household Income in the Past 12 Months (2022 Inflation Adj.):
+        - B29004_001E, ACS 2022 5Y
+    - Bachelor's Degree:
+        - B15003_022E, ACS 2022 5Y
+    - Age (???) -> Thinking about walking distance to bus stops for the elderly
+    - Disability (???) -> Accessibility issues, as above
+        
+"""
+
+################################################################################
+# SETUP 
+################################################################################
+
+#Dependencies
+import os
+import sys
+import csv
+import logging
+import requests
+# import pandas as pd
+from dotenv import load_dotenv
+from typing import List, Dict, Tuple
+
+#Environment variables, logging
+load_dotenv()
+CENSUS_API_KEY = os.getenv('CENSUS_API_KEY')
+logging.basicConfig(
+    level=logging.DEBUG, 
+    format='%(asctime)s - %(levelname)s - %(message)s'
+    )
+
```

Comment:
this works for now, down the road we may want a central place to track these (e.g. `utils/locations.py`) if they are used in many places as we would expand the cities we are tracking.